teamcity/pytest_plugin.py

```diff
@@ -103,13 +105,17 @@ def pytest_configure(config):
         output_capture_enabled = getattr(config.option, 'capture', 'fd') != 'no'
         coverage_controller = _get_coverage_controller(config)
         skip_passed_output = bool(config.getini('skippassedoutput'))
+        generate_test_suites = bool(config.getini('generatetestsuites'))
+        unmodified_verbosity = bool(config.getini('unmodifiedverbosity'))
 
-        config.option.verbose = 2  # don't truncate assert explanations
+        if not unmodified_verbosity:
+            config.option.verbose = 2  # don't truncate assert explanations
```

Comment:
I had to spend some time trying to understand why my verbosity control didn't work via command line arguments passed to `pytest`. I think, it's generally a bad idea to modify config object which is intended to retrieve data from it. To not break the API I decided to add parameter to turn this off. Can create a separate issue for that if it would be more convenient.